clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/CloudFoundryClientUtils.java

```diff
@@ -43,6 +43,8 @@ static void safelyCall(RetrofitConsumer r) throws CloudFoundryApiException {
       return Optional.of(r.call());
     } catch (RetrofitError retrofitError) {
       if (retrofitError.getResponse() != null && retrofitError.getResponse().getStatus() == 404) {
+        // FIXME: The oauthInterceptor could use a misconfigured endpoint and return 404 and
```

Comment:
Did you intend to provide an alternative in this PR?

Reply:
Tried to but that quickly seemed to require lots of change that'd be worth its own PR.

The main issue is a lack of context on why and what failed for the user.

My commit message is actually false and misleading, just tried it and an error is raised but there's just no details about it: `Cloud Foundry API returned with error(s): Unable to retrieve
  spaces.`
The comment in the code stand true.

Opted for a comment in the code in this PR because:
* Found the issue while testing the ssl-cert validation skip feature
* Mostly an edge case.
* Somehow mitigated by the WARN log message added in `HttpCloudFoundryClient` (same commit)
* Would require lots of effort for what seems like little value.

Reply:
Thanks @pdelagrave.